code/mob/living/critter/ai/flock/flocktasks.dm

```diff
@@ -136,10 +136,10 @@ stare
 	add_task(holder.get_instance(/datum/aiTask/succeedable/replicate, list(holder)))
 
 /datum/aiTask/sequence/goalbased/flock/replicate/precondition()
-	. = FALSE
 	var/mob/living/critter/flock/drone/F = holder.owner
-	if(F?.can_afford(FLOCK_LAY_EGG_COST + clamp((F.flock.getComplexDroneCount() - FLOCK_MIN_DESIRED_POP) * FLOCK_ADDITIONAL_RESOURCE_RESERVATION_PER_DRONE, 0, FLOCK_LAY_EGG_COST * 2)))
-		. = TRUE
+	if (!F.flock)
```

Comment:
Some weird nullchecking going on here- line 140 will runtime if `F == null`, but you still nullcheck it on line 142. Similarly we return on 141 if `F.flock == null`, but `F.flock` still gets null checked below.

tldr `if (!F?.flock)` and remove the others

Reply:
similar thing on line 200

Reply:
frankencode